src/Plugin/Field/FieldWidget/InlineEntityFormMultiple.php

```diff
@@ -418,14 +418,16 @@ public function formElement(FieldItemListInterface $items, $delta, array $elemen
     // field is required and empty, and there's only one allowed bundle).
     $entities = $form_state->get(['inline_entity_form', $this->getIefId(), 'entities']);
     if (empty($entities)) {
-      if (count($settings['handler_settings']['target_bundles']) == 1 && $this->fieldDefinition->isRequired() && !$this->settings['allow_existing']) {
-        $bundle = reset($settings['handler_settings']['target_bundles']);
+      if (count($target_bundles) == 1 && $this->fieldDefinition->isRequired() && !$this->settings['allow_existing']) {
+        $bundle = reset($target_bundles);
 
         // The parent entity type and bundle must not be the same as the inline
         // entity type and bundle, to prevent recursion.
-        if ($element['#entity_type'] != $settings['target_type'] || $element['#bundle'] != $bundle) {
-          $form_state['inline_entity_form'][$this->getIefId()]['form'] = 'add';
-          $form_state['inline_entity_form'][$this->getIefId()]['form settings'] = array(
+        $parent_entity_type = $form_state->getStorage()['form_display']->getTargetEntityTypeId();
+        $parent_bundle = $form_state->getStorage()['form_display']->getTargetBundle();
```

Comment:
No, we haven't. $element is a fieldset type created upper in the same function, and all informations we have is entities loaded with settings['target_type'] as '#entity_type'.

$form_state->getStorage()['form_display'] is the only way i found to get current entity_type and so avoid loading same entity_type and bundle.